tests/schemas/valid/test_placeholder_transform.json

```diff
@@ -107,6 +107,31 @@
                                 }
                               }
                             ]
+                          },
+                          {
+                            "placeholder": "generate_date_range",
+                            "transforms": [
+                              {
+                                "transform": "generate_date_range",
+                                "arguments": {
+                                  "reference_date": "2021-09-10",
+                                  "offset_full_weeks": -1,
+                                  "days_in_range": 7,
+                                  "first_day_of_week": "monday"
+                                }
+                              }
+                            ]
+                          },
+                          {
+                            "placeholder": "format_date_range_pair",
+                            "transforms": [
+                              {
+                                "transform": "format_date_range_pair",
+                                "arguments": {
+                                  "date_range": ["2021-09-10", "2021-09-15"]
+                                }
+                              }
+                            ]
                           }
```

Comment:
The schema should define a realistic schema, as we would never have this at run time, we should use a value source of the `reference_date` and 2 values sources for the format date range.

Reply:
I see, I was again using existing patterns as a guide - just above this section is the `email_links` transform which hardcodes some arguments. Is that a different use case? If so, should I be doing this:
```
"reference_date": {
                "$ref": "https://eq.ons.gov.uk/string_interpolation/definitions.json#/value_sources"
              }
```

Reply:
Never mind, I found another example to use for this 👍